Unreliable segmentation fault in TransformToIqt doctest

**Describe the bug**
There is an unreliable segmentation fault when we run the `TransformToIqt` usage example in the doctests. We have temporarily commented this example out while we investigate the problem #35991

**To Reproduce**
It is difficult to repeat the problem. Your best bet to reproduce it is to undo the changes made in #35991 , open a PR, and then use the `build_package_from_branch` pipeline to run the doctests on linux.

**Stacktrace**
```
Document: algorithms/TransformToIqt-v1
--------------------------------------
FileFinder-[Information] found path = /jenkins_workdir/workspace/build_packages_from_branch/source/build/ExternalData/Testing/Data/UnitTest/irs26176_graphite002_red.nxs
Load-[Notice] Load started
Load-[Information] Algorithm: Load v1
Load-[Information] Execution Date: 2023-08-24 10:07:23.761166000
Load-[Information] Execution Duration: -1 seconds
Load-[Information] UUID: 70585e01-93ac-4b89-995f-ce7844caf600
Load-[Information] Parameters:
Load-[Information]   Name: Filename, Value: /jenkins_workdir/ ... aphite002_red.nxs, Default?: No, Direction: Input
Load-[Information]   Name: OutputWorkspace, Value: sample, Default?: No, Direction: Output
Load-[Information]   Name: LoaderName, Value: LoadNexusProcessed, Default?: No, Direction: Output
Load-[Information]   Name: LoaderVersion, Value: 2, Default?: No, Direction: Output
Load-[Information]   Name: SpectrumMin, Value: 1, Default?: Yes, Direction: Input
Load-[Information]   Name: SpectrumMax, Value: 2147483647, Default?: Yes, Direction: Input
Load-[Information]   Name: SpectrumList, Value: , Default?: Yes, Direction: Input
Load-[Information]   Name: EntryNumber, Value: 0, Default?: Yes, Direction: Input
Load-[Information]   Name: LoadHistory, Value: 1, Default?: Yes, Direction: Input
Load-[Information]   Name: FastMultiPeriod, Value: 1, Default?: Yes, Direction: Input
Load-[Information] Using LoadNexusProcessed version 2.
InstrumentDefinitionParser-[Warning] Attribute 'mark-as' is a depricated attribute in Instrument Definition File. Please see the deprecated section of docs.mantidproject.org/concepts/InstrumentDefinitionFile for how to remove this warning message
InstrumentDefinitionParser-[Warning] Attribute 'mark-as' is a depricated attribute in Instrument Definition File. Please see the deprecated section of docs.mantidproject.org/concepts/InstrumentDefinitionFile for how to remove this warning message
InstrumentDefinitionParser-[Information] Loading geometry cache from /home/abc/.mantid/instrument/geometryCache/IRIS615cd1ff9686f1a61ab348fb9c4f3b32a99d1c36.vtp
Load-[Notice] Load successful, Duration 0.08 seconds
FileFinder-[Information] found path = /jenkins_workdir/workspace/build_packages_from_branch/source/build/ExternalData/Testing/Data/UnitTest/irs26173_graphite002_red.nxs
Load-[Notice] Load started
Load-[Information] Algorithm: Load v1
Load-[Information] Execution Date: 2023-08-24 10:07:23.878116000
Load-[Information] Execution Duration: -1 seconds
Load-[Information] UUID: 92d04f1e-50c0-4378-95d2-468a9bbab757
Load-[Information] Parameters:
Load-[Information]   Name: Filename, Value: /jenkins_workdir/ ... aphite002_red.nxs, Default?: No, Direction: Input
Load-[Information]   Name: OutputWorkspace, Value: can, Default?: No, Direction: Output
Load-[Information]   Name: LoaderName, Value: LoadNexusProcessed, Default?: No, Direction: Output
Load-[Information]   Name: LoaderVersion, Value: 2, Default?: No, Direction: Output
Load-[Information]   Name: SpectrumMin, Value: 1, Default?: Yes, Direction: Input
Load-[Information]   Name: SpectrumMax, Value: 2147483647, Default?: Yes, Direction: Input
Load-[Information]   Name: SpectrumList, Value: , Default?: Yes, Direction: Input
Load-[Information]   Name: EntryNumber, Value: 0, Default?: Yes, Direction: Input
Load-[Information]   Name: LoadHistory, Value: 1, Default?: Yes, Direction: Input
Load-[Information]   Name: FastMultiPeriod, Value: 1, Default?: Yes, Direction: Input
Load-[Information] Using LoadNexusProcessed version 2.
Load-[Notice] Load successful, Duration 0.06 seconds
TransformToIqt-[Notice] TransformToIqt started
TransformToIqt-[Information] Algorithm: TransformToIqt v1
TransformToIqt-[Information] Execution Date: 2023-08-24 10:07:23.941486000
TransformToIqt-[Information] Execution Duration: -1 seconds
TransformToIqt-[Information] UUID: 6da78f2f-0448-4d2f-a7f9-8f7643ad973a
TransformToIqt-[Information] Parameters:
TransformToIqt-[Information]   Name: SampleWorkspace, Value: sample, Default?: No, Direction: Input
TransformToIqt-[Information]   Name: ResolutionWorkspace, Value: can, Default?: No, Direction: Input
TransformToIqt-[Information]   Name: EnergyMin, Value: -0.5, Default?: Yes, Direction: Input
TransformToIqt-[Information]   Name: EnergyMax, Value: 0.5, Default?: Yes, Direction: Input
TransformToIqt-[Information]   Name: BinReductionFactor, Value: 10, Default?: Yes, Direction: Input
TransformToIqt-[Information]   Name: NumberOfIterations, Value: 50, Default?: Yes, Direction: Input
TransformToIqt-[Information]   Name: SeedValue, Value: 89631139, Default?: Yes, Direction: Input
TransformToIqt-[Information]   Name: ParameterWorkspace, Value: params, Default?: No, Direction: Output
TransformToIqt-[Information]   Name: OutputWorkspace, Value: iqt, Default?: No, Direction: Output
TransformToIqt-[Information]   Name: DryRun, Value: 0, Default?: Yes, Direction: Input
TransformToIqt-[Information]   Name: CalculateErrors, Value: 1, Default?: Yes, Direction: Input
Python-[Notice] Property deltaE-mode not found

Python-[Notice] Property deltaE-mode not found

Python-[Information] Analyser is graphite, reflection 002
Python-[Warning] Could not get the resolution from the IPF, using 1% of the E Fixed value for the resolution: 0.01845
CreateEmptyTableWorkspace-[Notice] CreateEmptyTableWorkspace started (child)
CreateEmptyTableWorkspace-[Information] Algorithm: CreateEmptyTableWorkspace v1
CreateEmptyTableWorkspace-[Information] Execution Date: 2023-08-24 10:07:23.946779000
CreateEmptyTableWorkspace-[Information] Execution Duration: -1 seconds
CreateEmptyTableWorkspace-[Information] UUID: ccc1f9df-1378-49b2-aa5e-dbb14d1e5618
CreateEmptyTableWorkspace-[Information] Parameters:
CreateEmptyTableWorkspace-[Information]   Name: OutputWorkspace, Value: params, Default?: No, Direction: Output
CreateEmptyTableWorkspace-[Notice] CreateEmptyTableWorkspace successful, Duration 0.00 seconds
Segmentation fault
ninja: build stopped: subcommand failed.
```

**Expected behavior**
There should be no segmentation fault, and the usage example should be uncommented.

**Platform/Version (please complete the following information):**
 - OS: Linux

--- a/docs/source/algorithms/TransformToIqt-v1.rst
+++ b/docs/source/algorithms/TransformToIqt-v1.rst
@@ -98,15 +98,22 @@ Usage
     sample = Load('irs26176_graphite002_red.nxs')
     can = Load('irs26173_graphite002_red.nxs')
 
-    # Commented out while we investigate an unreliable segmentation fault
-    # params, iqt = TransformToIqt(SampleWorkspace=sample,
-    #                              ResolutionWorkspace=can,
-    #                              EnergyMin=-0.5,
-    #                              EnergyMax=0.5,
-    #                              BinReductionFactor=10)
+    params, iqt = TransformToIqt(SampleWorkspace=sample,
+                                 ResolutionWorkspace=can,
+                                 EnergyMin=-0.5,
+                                 EnergyMax=0.5,
+                                 BinReductionFactor=10,
+                                 EnableLogging=True)
 
-    # print('Number of output bins: %d' % (params.cell('SampleOutputBins', 0)))
-    # print('Resolution bins: %d' % (params.cell('ResolutionBins', 0)))
+    print('Number of output bins: %d' % (params.cell('SampleOutputBins', 0)))
+    print('Resolution bins: %d' % (params.cell('ResolutionBins', 0)))
+
+Output:
+
+.. testoutput:: exTransformToIqtIRIS
+
+    Number of output bins: 172
+    Resolution bins: 6
 
 .. categories::
 

